lara-typescript/src/page-settings/components/page-settings-dialog.tsx

```diff
@@ -0,0 +1,164 @@
+import * as React from "react";
+import { Modal, ModalButtons } from "../../shared/components/modal/modal";
+import { Save } from "../../shared/components/icons/save-icon";
+import { Close } from "../../shared/components/icons/close-icon";
+
+import "./page-settings-dialog.scss";
+
+export interface IPageSettingsDialogProps {
+  title: string | undefined;
+  isHidden?: boolean;
+  isCompletion?: boolean;
+  hasArgBlock?: boolean;
+  hasStudentSidebar?: boolean;
+  hasTESidebar?: boolean;
+  updateSettingsFunction: (
+    updatedTitle: string | undefined,
+    updatedIsCompletion: boolean,
+    updatedIsHidden: boolean,
+    updatedHasArgBlock: boolean,
+    updatedHasStudentSidebar: boolean,
+    updatedHasTESidebar: boolean
+  ) => void;
+  closeDialogFunction: () => void;
+}
+
+export const PageSettingsDialog: React.FC<IPageSettingsDialogProps> = ({
+  title,
+  isHidden = false,
+  isCompletion = false,
+  hasArgBlock = false,
+  hasStudentSidebar = false,
+  hasTESidebar = false,
+  updateSettingsFunction,
+  closeDialogFunction
+  }: IPageSettingsDialogProps) => {
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    title = event.target.value;
+  };
+
+  const handleIsCompletionChange = () => {
+    isCompletion = !isCompletion;
+  };
+
+  const handleIsHiddenChange = () => {
+    isHidden = !isHidden;
+  };
+
+  const handleHasArgBlockChange = () => {
+    hasArgBlock = !hasArgBlock;
+  };
+
+  const handleHasStudentSidebarChange = () => {
+    hasStudentSidebar = !hasStudentSidebar;
+  };
+
+  const handleHasTESidebarChange = () => {
+    hasTESidebar = !hasTESidebar;
+  };
+
+  const handleUpdateSettings = () => {
+    console.log(title);
+    updateSettingsFunction(
```

Comment:
I am going to checkout this code locally and look closer, but is there an example of passing this property somewhere?

Reply:
I found it -- sorry.